System.Compiler/Reader.cs

```diff
@@ -2129,7 +2129,9 @@ private TypeNode GetTypeFromTypeName(SerializedTypeName typeName)
         string lookupName = localInfo.Name;
         bool hasNoPDBName = String.IsNullOrEmpty(lookupName);
 #if !FxCop
-        string name = hasNoPDBName ? "local"+i : lookupName;
+        // _ANONYM suffix will help to distinquish real locals from locals 
```

Comment:
:exclamation: I would prefer this be implemented by adding a new property to the `Local` class, similar to and located right next to `Local.HasNoPDBInfo`, which tracks the value of `hasNoPDBName`.